From 7a2c16a75836948e1172e82edee8ea9e571c0c2a Mon Sep 17 00:00:00 2001
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Mon, 17 Jan 2022 17:55:17 +0100
Subject: [PATCH 44/N] path_conv: a directory with `.lnk` suffix is not a
 `.lnk` file

When trying to create a directory called `xyz` in the presence of a
directory `xyz.lnk`, the MSYS2 runtime errors out with an `ENOENT`.

The root cause is actually a bit deeper: the `symlink_info::check()`
method tries to figure out whether the given path refers to a symbolic
link as emulated via `.lnk` files, but since it is a directory, that is
not the case. However, the `fileattr` field is not cleared, so that a
later `.exists()` call on the instance mistakenly thinks that the
symlink actually exists.

This fixes https://github.com/msys2/msys2-runtime/issues/81

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 winsup/cygwin/path.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/winsup/cygwin/path.cc b/winsup/cygwin/path.cc
index 36f15bd..ca65127 100644
--- a/winsup/cygwin/path.cc
+++ b/winsup/cygwin/path.cc
@@ -3498,6 +3498,7 @@ restart:
 	 hasn't been found. */
       if (ext_tacked_on && !had_ext && (fileattr & FILE_ATTRIBUTE_DIRECTORY))
 	{
+	  fileattr = INVALID_FILE_ATTRIBUTES;
 	  set_error (ENOENT);
 	  continue;
 	}
-- 
2.34.1

